### PR TITLE
Added valve opening and closing configuration for Sonoff TRVZB, v1.1.4 only

### PR DESCRIPTION
**Pull request by lcheng33775823**

### CODE REVIEW

File: src/devices/sonoff.ts [hunk 1]
```diff
@@ -622,6 +622,34 @@ const definitions: Definition[] = [
...
+                    'If the opening degree is set to 0%, the valve is fully closed when it is opened, ' +
+                    'and the default value is 100%, ' +
+                    'The valve opening degree should be greater than or equal to the valve closing degree.',
+                valueMin: 0.0,
+                valueMax: 100.0,
+                valueStep: 1.0,
+                unit: '%',
+            }),
+            numeric({
+                name: 'valve_closing_degree(version >= v1.1.4)',
```
Comment (Koenkk) [inline src/devices/sonoff.ts:640]: Please move `(version >= v1.1.4)` to the `description` (same for `valve_opening_degree`)

**Comment by PrenommeNomme** [comment]
Hello,

Do you think that the possibility of controlling the opening of the valve in percentage could be integrated into 1.1.5 version?

Thank you

**Comment by jamesonuk** [comment]
Just to say that I have just tried to apply this locally and I have the error

> 2024-04-08 17:29:36Publish 'get' 'valve opening' to 'Study TRV' failed: 'Error: Cluster with name '64529' does not exist'

which I think has been noted a few times in the 1.36.1 release thread so I don't think it is anything to do with this change but it is not working

**Comment by PrenommeNomme** [comment]
I thank you for your response and apologize for my messages which may be very approximate, as they are in automatic translation.  I would like to be sure of your answer, if I have interpreted it correctly, you are explaining to me that whatever the future or past version it will not work with this thermostatic valve.  I want to make sure, because I bought 8 of these thermostatic valves and I only have a short time to ask for a refund.  If I am not mistaken about all of this, would you be kind enough to recommend one or more brands and models of thermostatic valves with which a maximum of read and write attributes are supported to control them, and especially the  attribute to control the opening and closing of the thermostatic valve in degrees?  Thank you so much.

**Comment by jamesonuk** [comment]
I have had these working great with 1.1.1 firmware and 1.36 Zigbee2MQTT.    There is an issuewith the very latest release (is not affecting their actual usage) but I think it will be fixed quite quickly.

I have had eight working fine for a few months

**Comment by PrenommeNomme** [comment]
Ok. Thank you very much.

File: src/devices/sonoff.ts (hunk 1 above)
Comment (lcheng33775823) [reply src/devices/sonoff.ts]: Thanks for your reminding, it has been modified

**Comment by Koenkk** [comment]
Thanks!

**Comment by fir3drag0n** [comment]
I am getting errors if I want to set the valve position:

```
Info 2024-04-24 10:23:30Configuring 'TRV-Küche'
Error 2024-04-24 10:23:30Publish 'set' 'valve_opening_degree' to 'TRV-Küche' failed: 'Error: ZCL command 0x0cae5ffffeb0dc77/1 sonoffTrvzbCluster.write({"valveOpeningDegree":50}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":4742,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE')'
Error 2024-04-24 10:23:32Exception while calling fromZigbee converter: Value is not a number, got undefined (undefined)}
Error 2024-04-24 10:23:32Failed to configure 'TRV-Küche', attempt 2 (Error: ZCL command 0x0cae5ffffeb0dc77/1 sonoffTrvzbCluster.read([0,24576,24578,24579,24580,24581,24582,24583], {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":4742,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE') at Endpoint.checkStatus (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:314:28) at Endpoint.zclCommand (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:769:26) at processTicksAndRejections (node:internal/process/task_queues:95:5) at Endpoint.read (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:400:29) at configure (/app/node_modules/zigbee-herdsman-converters/src/devices/sonoff.ts:911:13) at Object.configure (/app/node_modules/zigbee-herdsman-converters/src/index.ts:160:21) at Configure.configure (/app/lib/extension/configure.ts:121:13) at EventEmitter.wrappedCallback (/app/lib/eventBus.ts:174:17))
Info 2024-04-24 10:23:43Configuring 'TRV-Küche'
Error 2024-04-24 10:23:43Publish 'set' 'valve_opening_degree' to 'TRV-Küche' failed: 'Error: ZCL command 0x0cae5ffffeb0dc77/1 sonoffTrvzbCluster.write({"valveOpeningDegree":55}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":4742,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE')'
Error 2024-04-24 10:23:44Received undefined command from '0'
Error 2024-04-24 10:23:45Exception while calling fromZigbee converter: Value is not a number, got undefined (undefined)}
Error 2024-04-24 10:23:45Failed to configure 'TRV-Küche', attempt 3 (Error: ZCL command 0x0cae5ffffeb0dc77/1 sonoffTrvzbCluster.read([0,24576,24578,24579,24580,24581,24582,24583], {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":4742,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE') at Endpoint.checkStatus (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:314:28) at Endpoint.zclCommand (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:769:26) at processTicksAndRejections (node:internal/process/task_queues:95:5) at Endpoint.read (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:400:29) at configure (/app/node_modules/zigbee-herdsman-converters/src/devices/sonoff.ts:911:13) at Object.configure (/app/node_modules/zigbee-herdsman-converters/src/index.ts:160:21) at Configure.configure (/app/lib/extension/configure.ts:121:13) at EventEmitter.wrappedCallback (/app/lib/eventBus.ts:174:17))
Error 2024-04-24 10:23:56Exception while calling fromZigbee converter: Value is not a number, got undefined (undefined)}
Error 2024-04-24 10:23:56Publish 'get' 'valve_opening_degree' to 'TRV-Küche' failed: 'Error: ZCL command 0x0cae5ffffeb0dc77/1 sonoffTrvzbCluster.read(["valveOpeningDegree"], {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":4742,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE')'
Error 2024-04-24 10:24:07Publish 'set' 'frost_protection_temperature' to 'TRV-Küche' failed: 'Error: ZCL command 0x0cae5ffffeb0dc77/1 sonoffTrvzbCluster.write({"frostProtectionTemperature":750}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":4742,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE')'
Error 2024-04-24 10:24:10Publish 'set' 'frost_protection_temperature' to 'TRV-Küche' failed: 'Error: ZCL command 0x0cae5ffffeb0dc77/1 sonoffTrvzbCluster.write({"frostProtectionTemperature":700}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":4742,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE')'
Error 2024-04-24 10:24:33Publish 'set' 'valve_opening_degree' to 'TRV-Küche' failed: 'Error: ZCL command 0x0cae5ffffeb0dc77/1 sonoffTrvzbCluster.write({"valveOpeningDegree":50}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":4742,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE')'
Error 2024-04-24 10:25:01Publish 'set' 'valve_opening_degree' to 'TRV-Küche' failed: 'Error: ZCL command 0x0cae5ffffeb0dc77/1 sonoffTrvzbCluster.write({"valveOpeningDegree":45}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":4742,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE')'
Error 2024-04-24 10:26:29Publish 'set' 'valve_opening_degree' to 'TRV-Küche' failed: 'Error: ZCL command 0x0cae5ffffeb0dc77/1 sonoffTrvzbCluster.write({"valveOpeningDegree":50}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":4742,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE')'
Error 2024-04-24 10:27:04Received undefined command from '0'
```

File: src/devices/sonoff.ts (hunk 1 above)
Comment (dhet) [reply src/devices/sonoff.ts]: The version of which component does this refer to exactly?

**Pull request merged**